test/mapmatch.cc

```diff
@@ -218,9 +218,13 @@ TEST(Mapmatch, test_matcher) {
     std::string walked_json;
 
     try {
+      Api api;
       walked_json = actor.trace_attributes(
           R"({"date_time":{"type":1,"value":"2019-10-31T18:30"},"costing":"auto","shape_match":"edge_walk","encoded_polyline":")" +
-          json_escape(encoded_shape) + "\"}");
+              json_escape(encoded_shape) + "\"}",
+          nullptr, &api);
+      EXPECT_NE(api.trip().routes(0).legs(0).location(0).path_edges_size(), 0);
+      EXPECT_NE(api.trip().routes(0).legs(0).location().rbegin()->path_edges_size(), 0);
```

Comment:
these and the 2 below were failing in the other PR #3274 without the fixes in this pr